### PR TITLE
feat: Add cta on billing for >20k annual spend customers

### PR DESCRIPTION
**Pull request by xrdt**
## Problem

Updates the existing >$1k monthly spend alert to include more perks we can offer to large customers. 

## Changes

No CTA: 
Screenshot 2023-12-22 at 10 41 54 AM

With CTA: 
Screenshot 2023-12-22 at 10 42 46 AM

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/billing/Billing.tsx [hunk 1]
```diff
@@ -237,38 +208,92 @@ export function Billing(): JSX.Element {
...
+                            <LemonCheckbox checked disabled label="Save 20% by switching to up-front annual billing" />
+                            <LemonCheckbox
+                                checked
+                                disabled
+                                label="Discounts for bundling subscriptions to multiple products"
+                            />
+                            <LemonCheckbox checked disabled label="Team training" />
+                            <LemonCheckbox checked disabled label="Dedicated support in a Slack channel" />
+                            <LemonCheckbox checked disabled label="Free merch" />
+                            <div style={{ alignSelf: 'start', paddingTop: '8px' }}>
```
Comment (xrdt) [inline frontend/src/scenes/billing/Billing.tsx:244]: Need to fix this lint issue with styling the lemonbutton.
Also see line 284

File: frontend/src/scenes/billing/Billing.tsx [hunk 2]
```diff
@@ -237,38 +208,92 @@ export function Billing(): JSX.Element {
...
+                                <LemonButton type="secondary" to="mailto:sales@posthog.com">
+                                    Get in touch!
```
Comment (xrdt) [inline frontend/src/scenes/billing/Billing.tsx:246]: We can collapse this once they click the button. Need to think about collapsed state.

Comment (xrdt) [reply frontend/src/scenes/billing/Billing.tsx:246]: Is it better to have an explicit dismiss button? Either something that appears on hover of the hog, say, or a x in the upper right that appears on hover?

Comment (xrdt) [reply frontend/src/scenes/billing/Billing.tsx:246]: Or a dismiss button next to the Get in touch <<< this is probably the simplest.

**Comment by raquelmsmith** [comment]
IMO this doesn't need to be dismissable, it can just be a persistent "ad"

**Comment by xrdt** [comment]
Newer screenshot 
Screenshot 2024-01-02 at 1 07 27 PM

**Comment by xrdt** [comment]
Re dismissability: Can hide the CTA once they've converted to an annual plan.

**Comment by raquelmsmith** [comment]
Changed how it looks a bit:

image

There's kind of an awkward gap above the "Products" header but cory wants to redo this page anyway so I think it's fine.

File: frontend/public/hedgehog/builder-hog-waving-smiling.png
Comment (raquelmsmith) [inline frontend/public/hedgehog/builder-hog-waving-smiling.png:1]: do we need this?

File: frontend/src/scenes/billing/billingLogic.ts [hunk 3]
```diff
@@ -67,6 +67,7 @@ export const billingLogic = kea<billingLogicType>([
         registerInstrumentationProps: true,
         setRedirectPath: true,
         setIsOnboarding: true,
+        setPerksCTADismissed: (perksCTADismissed: boolean) => ({ perksCTADismissed }),
```
Comment (raquelmsmith) [inline frontend/src/scenes/billing/billingLogic.ts:70]: I don't think we need this action, can just keep it there until they sign up for annual.

File: frontend/public/hedgehog/builder-hog-waving-smiling.png
Comment (raquelmsmith) [inline frontend/public/hedgehog/builder-hog-waving-smiling.png:1]: Do we need this file?

File: frontend/src/lib/components/hedgehogs.tsx [hunk 4]
```diff
@@ -140,3 +141,6 @@ export const MicrophoneHog = (props: HedgehogProps): JSX.Element => {
 export const PhonePairHogs = (props: HedgehogProps): JSX.Element => {
     return <SquaredHedgehog src={phonePairHogs} {...props} />
 }
+export const BuilderHogWavingSmiling = (props: HedgehogProps): JSX.Element => {
+    return <SquaredHedgehog src={builderHogWavingSmiling} {...props} />
+}
```
Comment (raquelmsmith) [inline frontend/src/lib/components/hedgehogs.tsx:146]: Do we not already have this one in there? Prob remove since not using it anyway.

**Review by raquelmsmith — CHANGES_REQUESTED** [review]
Just removing the dismiss code and the extra hog!

File: frontend/__snapshots__/scenes-other-billing-v2--billing-v-2-with-discount--dark.png
Comment (raquelmsmith) [inline frontend/__snapshots__/scenes-other-billing-v2--billing-v-2-with-discount--dark.png:1]: 👀 doesn't look very good in dark mode, worth a quick color pass

**Review by raquelmsmith — CHANGES_REQUESTED** [review]
dark mode!

File: frontend/__snapshots__/scenes-other-billing-v2--billing-v-2-with-discount--dark.png
Comment (xrdt) [reply frontend/__snapshots__/scenes-other-billing-v2--billing-v-2-with-discount--dark.png:1]: Good catch, thanks.

Comment (xrdt) [reply frontend/__snapshots__/scenes-other-billing-v2--billing-v-2-with-discount--dark.png:1]: Screenshot 2024-01-03 at 2 06 03 PM

**Review by raquelmsmith — APPROVED** [review]

**Pull request merged**